### ISSUE TITLE
EI_EXPOSE_REP false positive when returning a `List.copyOf` / `Set.copyOf` result

### ISSUE THREAD

**Issue by Emilyaxe** [spotbugs/spotbugs#4033]
**Describe the bug**
SpotBugs reports `EI_EXPOSE_REP` for getters returning fields initialized with `List.copyOf(...)` or `Set.copyOf(...)`, even though these JDK methods return unmodifiable collections. It correctly suppresses the warning for `List.of(...)` / `Set.of(...)`, but fails to treat `copyOf` as equally immutable, leading to a false positive.

**To Reproduce**

```java
import java.util.List;
import java.util.Set;

public class CollectionHolder {

    // BEFORE — no warning
    private static final List<String> NAMES_OK = List.of("a", "b", "c");
    private static final Set<Integer>  CODES_OK = Set.of(1, 2, 3);

    // AFTER — EI_EXPOSE_REP reported on getNames() / getCodes()
    private static final List<String> NAMES = List.copyOf(List.of("a", "b", "c"));
    private static final Set<Integer>  CODES = Set.copyOf(Set.of(1, 2, 3));
    private final        List<String> tags  = List.copyOf(List.of("x", "y", "z"));

    public List<String> getNames() { return NAMES; }   // flagged
    public Set<Integer> getCodes() { return CODES; }   // flagged
    public List<String> getTags()  { return tags;  }   // flagged
}
```

Both versions return the same immutable JDK collection types (`ImmutableCollections.ListN` / `SetN`), yet only the `copyOf`-initialized version triggers `EI_EXPOSE_REP`.

**Expected behavior**
No `EI_EXPOSE_REP` warning: `List.copyOf` and `Set.copyOf` (and `Map.copyOf`) are contractually guaranteed to return unmodifiable collections, so the getter cannot expose mutable internal state.

**Actual behavior**
`EI_EXPOSE_REP` (priority 2) is raised on each getter whose backing field is initialized via `List.copyOf` / `Set.copyOf`, while the structurally equivalent `List.of` / `Set.of` case is correctly ignored.

**Environment**

- Java version: JDK 21 (Eclipse Temurin)
- SpotBugs Version: 4.9.8
- How do you use SpotBugs: CLI

**Comment by timothyjward** [comment]
We see similar issues in Eclipse sensiNact for the following case - we have records which permit null, or take copies of Lists. These complain that `bbox` and `properties` may expose internals, when in fact they don't. Strangely `foreignMembers` isn't a problem.

```java
/*********************************************************************
* Copyright (c) 2025 Contributors to the Eclipse Foundation.
*
* This program and the accompanying materials are made
* available under the terms of the Eclipse Public License 2.0
* which is available at https://www.eclipse.org/legal/epl-2.0/
*
* SPDX-License-Identifier: EPL-2.0
*
* Contributors:
*   Kentyou - initial implementation
*   Tim Ward - refactor as records
**********************************************************************/
package org.eclipse.sensinact.gateway.geojson;

import java.util.List;
import java.util.Map;

import org.jspecify.annotations.Nullable;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonInclude;

public record Feature(@Nullable String id, @Nullable @JsonInclude Geometry geometry, 
        @Nullable @JsonInclude Map<String, Object> properties, @Nullable List<Double> bbox,
        @JsonAnySetter @JsonAnyGetter @Nullable Map<String,Object> foreignMembers) implements GeoJsonObject {

    public Feature {
        if(bbox != null) {
            bbox = List.copyOf(bbox);
        }
        if(properties != null) {
            properties = Map.copyOf(properties);
        }
        if(foreignMembers != null) {
            foreignMembers = Map.copyOf(foreignMembers);
        } else {
            foreignMembers = Map.of();
        }
    }

    @Override
    public GeoJsonType type() {
        return GeoJsonType.Feature;
    }

    @Override
    public boolean isEmpty() {
        return geometry == null || geometry.isEmpty();
    }
}
```

Another, more complex example is found in nested lists, where we we stream the list.

```java
/*********************************************************************
* Copyright (c) 2025 Contributors to the Eclipse Foundation.
*
* This program and the accompanying materials are made
* available under the terms of the Eclipse Public License 2.0
* which is available at https://www.eclipse.org/legal/epl-2.0/
*
* SPDX-License-Identifier: EPL-2.0
*
* Contributors:
*   Kentyou - initial implementation
*   Tim Ward - refactor as records
**********************************************************************/
package org.eclipse.sensinact.gateway.geojson;

import java.util.List;
import java.util.Map;

import org.jspecify.annotations.Nullable;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;

/**
 * A GeoJSON multi polygon object as defined in
 * <a href="https://tools.ietf.org/html/rfc7946#section-3.1">the GeoJSON
 * specification</a>
 */
public record MultiPolygon(List<List<List<Coordinates>>> coordinates, @Nullable List<Double> bbox,
        @JsonAnySetter @JsonAnyGetter @Nullable Map<String,Object> foreignMembers) implements Geometry {

    public MultiPolygon {
        if(coordinates != null) {
            coordinates = coordinates.stream()
                    .map(l -> l.stream().map(List::copyOf).toList())
                    .toList();
        } else {
            // GeoJSON specification 3.1 - GeoJSON processors MAY interpret Geometry objects with
            // empty "coordinates" arrays as null objects.
            coordinates = List.of();
        }
        if(bbox != null) {
            bbox = List.copyOf(bbox);
        }
        if(foreignMembers != null) {
            foreignMembers = Map.copyOf(foreignMembers);
        } else {
            foreignMembers = Map.of();
        }
    }

    @Override
    public GeoJsonType type() {
        return GeoJsonType.MultiPolygon;
    }

    @Override
    public boolean isEmpty() {
        return coordinates.isEmpty();
    }
}
```
